Merge tests under tracing / Reflection / ReadyToRun / Regressions

trylek opened the pull request.

trylek committed:
Manual fixes to tests under tracing

Includes deduplication of two "common" libraries - one under the
folder "tracing/common" (renamed to tracing_common) and another
under "tracing/eventpipe/common" (renamed to eventpipe_common).

Thanks

Tomas

trylek committed:
Mechanically merge tests under the folder "tracing"

trylek committed:
Manual adjustments for tests under "reflection"

trylek committed:
Mechanically merge all tests under "reflection"

trylek committed:
Manual adjustments to tests under readytorun

trylek committed:
Mechanically merge tests under readytorun

trylek committed:
Move coreroot_determinism to a separate wrapper as it's a lengthy test

trylek committed:
Manual adjustments of tests under Regressions

trylek committed:
Mechanically merge the Regressions folder

trylek committed:
Fix a previously missed property under readytorun/D.B.props

markples commented on src/tests/Regressions/coreclr/0416/hello.cs:
```diff
@@ -1,15 +1,16 @@
+        Console.WriteLine("  DOTNET_LogEnable=1");
+        Console.WriteLine("  DOTNET_LogToConsole=1");
+        Console.WriteLine("This test ensures that logging is working");
+    }
```
This test doesn't seem to be doing that.

This isn't related to your change except that I was going to comment that the comment for RPI should indicate the environment variables, except...

(I wouldn't hold this change up - just file an item or even just delete this test)

trylek replied:
Thanks Mark for spotting, I do admit I haven't scrutinized every single test but this one does look weird. I'm going to push a commit deleting it and, if someone complains, we can bring it back and possibly try to make sure it does something reasonable.

trylek committed:
Delete meaningless test per Mark's PR feedback

markples commented on src/tests/Regressions/coreclr/22021/provider.ilproj:
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk.IL">
   <PropertyGroup>
     <OutputType>Library</OutputType>
+    <CLRTestKind>BuildOnly</CLRTestKind>
```
I think D.B.targets will now automatically set this to `SharedLibrary` - did you hit a problem with that?

trylek replied:
Thanks for pointing that out, I tried to rebuild the tests without the property and it works fine, in fact I could remove both properties as I believe OutputType is Library by default. Fixed in 15th commit.

markples commented on src/tests/readytorun/coreroot_determinism/coreroot_determinism.csproj:
```diff
@@ -1,5 +1,7 @@
   <PropertyGroup>
+    <!-- Needed for CLRTestTargetUnsupported, GCStressIncompatible, CrossGenTest -->
+    <RequiresProcessIsolation>true</RequiresProcessIsolation>
     <OutputType>exe</OutputType>
```
There are several lowercase exes in this batch that got left behind like before.

trylek replied:
Fixed in 12th commit.

markples reviewed (approved):
Thank you for continuing to push on this!

trylek committed:
Remove unneeded exe output types per Mark's PR feedback

trylek committed:
Fix the renamed eventpipe_common project name in one JIT directed test

trylek committed:
Fix two more tests referencing the renamed eventpipe_common project

trylek committed:
Remove superfluous BuildOnly property from the provider.ilproj project

trylek commented:
/azp run runtime-coreclr outerloop

trylek committed:
Fix compilation failure in the 22888 regression test

This failure was caused by incorrectly making a test out of what
is in reality a library used by custom means by the test app
(for the purpose of accessing resources). While I believe we should
further clean this logic up, this should be good enough to make
tests pass.

Thanks

Tomas

trylek committed:
Fix two readytorun tests to work in the merged mode

trylek committed:
Fix mainv1 and mainv2 version resiliency tests

trylek committed:
Fix missing dependency of the mainv2 test on testv2

trylek commented:
/azp run runtime-coreclr outerloop

trylek commented:
/azp run runtime-extra-platforms
